lineapy/plugins/session_writers.py

```diff
@@ -0,0 +1,238 @@
+from typing import Dict, List
+
+from lineapy.graph_reader.node_collection import (
+    ArtifactNodeCollection,
+    UserCodeNodeCollection,
+)
+from lineapy.graph_reader.session_artifacts import SessionArtifacts
+from lineapy.graph_reader.types import InputVariable
+from lineapy.plugins.utils import load_plugin_template
+
+
+class BaseSessionWriter:
+    """
+    BaseSessionWriter contains helper functions to turn the various components of a SessionArtifacts
+    object to runnable code.
+    """
+
+    def __init__(self):
+        pass
+
+    def get_session_module_imports(
+        self,
+        session_artifact: SessionArtifacts,
+        include_non_slice_as_comment=False,
+        indentation=0,
+    ) -> str:
+        """
+        Return all the import statement for the session.
+        """
+
+        return session_artifact.import_nodecollection.get_import_block(
+            graph=session_artifact.graph,
+            include_non_slice_as_comment=include_non_slice_as_comment,
```

Comment:
This i hardcoded to be false in the prev version to keep the extra comments out of the beginning of the file. pretty much only bothered with keeping extra things inside of the function defs that are produced.

Reply:
wait so we aren't providing all the imports from the session in comments? I can comment this out and you can add it back in if you think thats a case we need to support.